File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/ProfileView.java
package app.editors.manager.mvp.views.main;

import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

import java.util.List;

import app.editors.manager.mvp.models.user.Thirdparty;
import app.editors.manager.mvp.views.base.BaseView;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface ProfileView extends BaseView {
    void onWebDavState();
    void onCloudState();
    void onOnlineState();
    void onEmptyThirdparty();
    void onSetServices(List<Thirdparty> list);
    void onAccountType(String type);
    void onClose();
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/ProfileView.kt
package app.editors.manager.mvp.views.main

import app.documents.core.account.CloudAccount
import app.editors.manager.mvp.presenters.main.ProfileState
import app.editors.manager.mvp.views.base.BaseView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface ProfileView : BaseView {
    fun onClose(isLogout: Boolean, account: CloudAccount? = null)
    fun onRender(state: ProfileState)
}